### ISSUE TITLE
Mypy flags `unreachable` on `pre-save` and checking for instance ID

### ISSUE THREAD

**Issue by sshishov** [typeddjango/django-stubs#2014]
# Bug report

We are implementing pre-save of some models and there usually we check if the instance is being created or not by doing
```python
if instance.id is None:
    ...  # creating the instance logic
```

## What's wrong

```python
import typing as t

from django.db import models as django_models
from django import dispatch as django_dispatch

@django_dispatch.receiver(django_models.signals.pre_save, sender=my_app.MyModel)
def project_pre_save(sender: type[my_app.MyModel], instance: my_app.MyModel, **kwargs: t.Any) -> None:
    if instance.id is None:
        add_something_to_instance(instance)  # <-- Statement is unreachable  [unreachable]
    ...
```

In this case mypy flagging this with the following error indicated above

## How is that should be

On pre-save `instance.id` should be either the Field or None as it can be None in case of creation of the object.
Do not know how to properly implement it though...

## System information

- OS: MacOS / Linux
- `python` version: 3.11.5
- `django` version: 3.2.23
- `mypy` version: 1.9.0
- `django-stubs` version: 4.2.7
- `django-stubs-ext` version: 4.2.7

**Comment by sobolevn** [comment]
Yes, this is the trade-off we had to make. We can add fake `UnsavedModel` subclass (to stubs and ext) of `Model` with `id` set to `T | None`

What do you think?

**Comment by sshishov** [comment]
Hi @sobolevn 

Thank you for the report, I just have a question how this `UnsavedModel` will be used by people. If it will require a lot of "hand-writing" or workarounds then better to keep it as is and use `type: ignore[<specific-error>]` instead.

Will it be like a wrapper? Like we have `WithAnnotations` for queryset/model?

If it will be like a wrapper, then I am "supporting" this approach!

Thanks for your valuable time! ⏲️
